src/LinuxCaptureService/LinuxCaptureService.cpp

```diff
@@ -113,17 +113,67 @@ static void StopInternalProducersAndCaptureStartStopListenersInParallel(
   }
 }
 
+namespace {
+
+// This class hijacks FunctionEntry and FunctionExit events before they reach the
+// ProducerEventProcessor, and sends them to LinuxTracing instead, so that they can be processed
+// like u(ret)probes. All the other events are forwarded to the ProducerEventProcessor normally.
+class ProducerEventProcessorHijackingFunctionEntryExitForLinuxTracing
+    : public ProducerEventProcessor {
+ public:
+  ProducerEventProcessorHijackingFunctionEntryExitForLinuxTracing(
+      ProducerEventProcessor* original_producer_event_processor, TracingHandler* tracing_handler)
+      : producer_event_processor_{original_producer_event_processor},
+        tracing_handler_{tracing_handler} {}
+  ~ProducerEventProcessorHijackingFunctionEntryExitForLinuxTracing() override = default;
+
+  void ProcessEvent(uint64_t producer_id,
+                    orbit_grpc_protos::ProducerCaptureEvent&& event) override {
+    switch (event.event_case()) {
+      case ProducerCaptureEvent::kFunctionEntry:
+        // TODO(b/194704608): Forward to LinuxTracing through TracingHandler.
+        LOG("FunctionEntry: pid=%u, tid=%u, function_id=%u, stack_pointer=%#x, return_address=%#x, "
+            "timestamp_ns=%u",
+            event.function_entry().pid(), event.function_entry().tid(),
+            event.function_entry().function_id(), event.function_entry().stack_pointer(),
+            event.function_entry().return_address(), event.function_entry().timestamp_ns());
+        (void)tracing_handler_;
+        break;
+      case ProducerCaptureEvent::kFunctionExit:
+        // TODO(b/194704608): Forward to LinuxTracing through TracingHandler.
+        LOG("FunctionExit: pid=%u, tid=%u, timestamp_ns=%u", event.function_exit().pid(),
+            event.function_exit().tid(), event.function_exit().timestamp_ns());
+        (void)tracing_handler_;
```

Comment:
Yes, that the field is unused. It was just for the intermediate commit, I guess that for this case it would have been less confusing to squash before the review.